CriticalMass/FollowFriendsViewController.swift

```diff
@@ -9,37 +9,62 @@ import UIKit
 
 class FollowFriendsViewController: UIViewController {
     private var urlString: String?
+    private var name: String
+    private var oldBrightness: CGFloat
+
+    init(name: String) {
+        self.name = name
+        oldBrightness = UIScreen.main.brightness
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        view.backgroundColor = .white
+        view.backgroundColor = .yellow100
+
+        configureNavigationBar()
 
         do {
             let publicKeyData = try RSAKey(tag: RSAKey.keychainTag).publicKeyDataRepresentation()
-            urlString = try FollowURLObject(queryObject: Friend(name: "TODO", key: publicKeyData)).asURL()
+            urlString = try FollowURLObject(queryObject: Friend(name: name, key: publicKeyData)).asURL()
         } catch {
             // TODO: present error
             fatalError()
         }
 
         configureQRCodeView()
-        configureShareLinkButton()
     }
 
-    private func configureShareLinkButton() {
-        let button = UIButton()
-        button.setTitle("Share Link", for: .normal)
-        button.setTitleColor(.blue, for: .normal)
-        button.addTarget(self, action: #selector(shareLinkButtonTapped), for: .touchUpInside)
-        button.sizeToFit()
-        button.center = view.center.applying(CGAffineTransform(translationX: 0, y: -200))
-        view.addSubview(button)
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        UIScreen.main.brightness = 100
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        UIScreen.main.brightness = oldBrightness
+    }
+
+    private func configureNavigationBar() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .action, target: self, action: #selector(shareLinkButtonTapped))
     }
 
     private func configureQRCodeView() {
-        let view = QRCodeView()
+        let backgroundView = QRCodeBackgroundView(frame: CGRect(origin: .zero, size: CGSize(width: 280, height: 280)))
+        backgroundView.center = self.view.center
```

Comment:
I'd prefer some NSLayoutConstraint but this is fine too

Reply:
Good point. Added an helper function for that